drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -1567,7 +1567,9 @@ void RasterizerCanvasGLES3::_add_to_batch(uint32_t &r_index, bool &r_batch_broke
 
 void RasterizerCanvasGLES3::_new_batch(bool &r_batch_broken) {
 	if (state.canvas_instance_batches.size() == 0) {
-		state.canvas_instance_batches.push_back(Batch());
+		Batch new_batch;
+		new_batch.instance_buffer_index = state.current_instance_buffer_index;
```

Comment:
We need to start at the last instance buffer, even when starting the first batch. This is 0 when starting a new canvas draw, but will be non-zero if there are multiple draws in a single canvas draw.